301-storage-spaces-direct/nested/deploy-s2d-cluster.json

```diff
@@ -0,0 +1,349 @@
+						"name": "ipconfig1",
+						"properties": {
+							"privateIPAllocationMethod": "Dynamic",
+							"subnet": {
+								"id": "[parameters('nicSubnetUri')]"
+							}
+						}
+					}
+				]
+			}
+		},
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2015-01-01",
+            "copy": {
+                "name": "virtualMachineLoop",
+                "count": "[parameters('vmCount')]"
+            },
+            "name": "[concat(variables('vmNamePrefix'),copyindex(),'-newVM')]",
+			"dependsOn": [
+				"storageAccountLoop",
+				"nicLoop",
+				"[resourceId('Microsoft.Compute/availabilitySets', variables('vmAvailabilitySetName'))]"
+			],
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[variables('vmLoopTemplateURL')]"
+                },
+                "parameters": {
+                    "vmName": {
+                        "value": "[concat(variables('vmNamePrefix'), copyindex())]"
+                    },
+                    "storageAccountUri": {
+                        "value": "[reference(concat('Microsoft.Storage/storageAccounts/',variables('StorageNamePrefix'),copyIndex()),variables('apiVersionStorage')).primaryEndpoints.blob]"
+                    },
+                    "vmAvailabilitySetName": {
+                        "value": "[variables('vmAvailabilitySetName')]"
+                    },
+                    "vmSize": {
+                        "value": "[parameters('vmSize')]"
+                    },
+                    "vmDiskCount": {
+                        "value": "[parameters('vmDiskCount')]"
+                    },
+                    "vmDiskSize": {
+                        "value": "[parameters('vmDiskSize')]"
+                    },
+					"vhdContainerName": {
+						"value": "[variables('vhdContainerName')]"
+					},
+                    "adminUsername": {
+                        "value": "[parameters('adminUsername')]"
+                    },
+                    "adminPassword": {
+                        "value": "[parameters('adminPassword')]"
+                    },
+                    "_artifactsLocation": {
+                        "value": "[parameters('_artifactsLocation')]"
+                    },
+                    "_artifactsLocationSasToken": {
+                        "value": "[parameters('_artifactsLocationSasToken')]"
+                    },													
+					"imagePublisher": {
+						"value": "[parameters('imagePublisher')]"
+					},
+					"imageOffer": {
+						"value": "[parameters('imageOffer')]"
+					},
+					"imageSKU": {
+						"value": "[parameters('imageSKU')]"
+					},
+					"imageVersion": {
+						"value": "[parameters('imageVersion')]"
+					}
+                }
+            }
+        },
+		{
+			"type": "Microsoft.Compute/virtualMachines/extensions",
+			"name": "[concat(variables('vmNamePrefix'),string(add(copyindex(),1)),'/s2dPrep')]",
```

Comment:
I think this can be simplified...

concat(variables('vmNamePrefix'), copyindex(1), '/s2dPrep')